hls4ml/model/optimizer/passes/merge_relu.py

```diff
@@ -0,0 +1,48 @@
+from hls4ml.model.optimizer import OptimizerPass
+from hls4ml.model.layers import Activation, Dense, Conv2D, Conv2DBatchnorm
+
+class MergeRelu(OptimizerPass):
+    def match(self, node):
+        supported_layers = (Dense, Conv2D, Conv2DBatchnorm)
+
+        is_match = issubclass(node.get_input_node().__class__, supported_layers)
+        # ReLU layers are of class Activation
+        is_match = is_match and issubclass(node.__class__, Activation)
+        return is_match
```

Comment:
I see. For the missing `CONFIG_T::out_t` error, it looks like this `match()` function is too generous. Since all activation layers are of subclass `Activation`, any Dense/Conv2D layer that is followed by any activation function returns True, which is wrong. I'll look into tightening this up.